src/txn-wal/src/txn_cache.rs

```diff
@@ -1011,20 +1117,26 @@ pub(crate) struct DataRegistered<T> {
     /// The inclusive time at which the data shard was removed from the txns
     /// set, or None if it hasn't yet been removed.
     pub(crate) forget_ts: Option<T>,
+
+    /// The Schema of this data shard's keys.
+    key_schema: KS,
+    /// The Schema of this data shard's vals.
+    val_schema: VS,
 }
 
-impl<T: Timestamp + TotalOrder> DataRegistered<T> {
-    pub(crate) fn contains(&self, ts: &T) -> bool {
-        &self.register_ts <= ts && self.forget_ts.as_ref().map_or(true, |x| ts <= x)
+impl<KS, VS, T: Timestamp + TotalOrder> DataRegistered<KS, VS, T> {
+    pub(crate) fn contains(&self, ts: &T) -> Option<(&KS, &VS)> {
```

Comment:
Might be worth adding a doc-comment to this since it might be surprising that this doesn't return `bool`.